### PR TITLE
Add SafeBase Module

### PR DESCRIPTION
**Pull request by 0x0f0f0f**
- **add safe base module**
- **Add SafeBase**
- **add more tests**

### CODE REVIEW

**Comment by 0x0f0f0f** [comment]
@iamed2 would you have a moment to take a look at this?

File: src/SafeBase.jl [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    in module `$(err.mod)`, caused by:
+
+    """,
+  )
+  Base.display_error(io, err.exc, err.bt)
+end
+
+
+"""
+Evaluate expression `expr` in module `m` (defaulting to `Kamchatka`).
```
Comment (iamed2) [inline src/SafeBase.jl]: Kamchatka?

File: src/SafeBase.jl [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+This function can optionally throw if an exception is encountered.
+Otherwise, a silent failure is indicated by returning `nothing`.
+"""
+function safe_eval(m::Module, expr)::Result{Any,EvalError}
```
Comment (iamed2) [inline src/SafeBase.jl]: What types can this accept? Why such a different implementation here?

File: src/SafeBase.jl [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  mod::Module
+  expr::Union{Expr,Symbol}
+  exc::Exception
+  bt::Backtrace
+end
+
+format_code(ex::Symbol) = repr(ex)
+
+function Base.showerror(io::IO, err::EvalError)
+  expression_str = replace(repr(err.expr), r"^"m => "  ")
```
Comment (iamed2) [inline src/SafeBase.jl:142]: Potentially misleading in output. Why do this?

File: src/SafeBase.jl [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+It also contains some specific errors that meant to be more informative.
+"""
+module SafeBase
+
+using JuliaSyntax
+using ..ResultTypes
+export ParseError, safe_parse, safe_parse_julia, safe_parseall_julia, parse_julia, parseall_julia, EvalError, safe_eval
+
+"A stack trace, as would be accessed when catching an exception."
+const Backtrace = Vector{Union{Base.InterpreterIP,Ptr{Nothing}}}
```
Comment (iamed2) [inline src/SafeBase.jl]: This seems like a replication of something internal, maybe it's better to just say `Any` than risk possible values changing in Julia internals.

**Comment by iamed2** [comment]
I can look more this weekend, in the meantime can you make src/SafeBase.jl more BlueStyle? And add signatures to the new method docstrings that don't contain them.

File: src/SafeBase.jl (hunk 1 above)
Comment (0x0f0f0f) [reply src/SafeBase.jl]: Leftover

File: src/SafeBase.jl (hunk 2 above)
Comment (0x0f0f0f) [reply src/SafeBase.jl]: Should be `Any`.

File: src/SafeBase.jl (hunk 3 above)
Comment (0x0f0f0f) [reply src/SafeBase.jl:142]: It's just for tabbing. `^` is start of line regex, i misread it as the char too. Why misleading? it just "tabs" the expression

**Comment by 0x0f0f0f** [comment]
@iamed2 addressed the comments. Thanks!

**Comment by 0x0f0f0f** [comment]
Can I go ahead and merge? @iamed2

**Pull request merged**